### PR TITLE
[FLINK-25731][connectors/kinesis] Deprecated FlinkKinesisConsumer / F…

### PR DESCRIPTION
**Pull request by CrynetLogistics**
…linkKinesisProducer

## What is the purpose of the change

To deprecate the FlinkKinesisConsumer / FlinkKinesisProducer in favour of:
 * This consumer based on the Kinesis Consumer Library (KCL) has been deprecated, a Jira to create an equivalent consumer based on AWS SDK for Java 2.x has been raised in FLINK-25729 and planned for Flink 1.16.

 * This producer based on the Kinesis Producer Library KPL has been superseded. The new sink can be found in the module flink-connectors/flink-connector-aws-kinesis-data-streams and package org.apache.flink.connector.kinesis.sink.KinesisDataStreamsSink. It is based on the AWS SDK for Java 2.x. The work to replace this sink was carried out in FLINK-24227.

## Brief change log

- Deprecated FlinkKinesisConsumer / FlinkKinesisProducer with message.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)n
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)n
  - The serializers: (yes / no / don't know)n
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)n
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)n
  - The S3 file system connector: (yes / no / don't know)n

## Documentation

  - Does this pull request introduce a new feature? (yes / no)n
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java [hunk 1]
```diff
@@ -101,7 +101,11 @@
  * source or a downstream operator.
  *
  * @param <T> the type of data emitted
+ * @deprecated This consumer based on the Kinesis Consumer Library (KCL) has been deprecated, a jira
+ *     to create an equivalent consumer based on AWS SDK for Java 2.x has been raised in FLINK-25729
+ *     and planned for Flink 1.16.
```
Comment (dannycranmer) [inline flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java:106]: Please remove the 1.16 reference, this is captured in the Jira

**Review by dannycranmer — CHANGES_REQUESTED** [review]

File: flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java (hunk 1 above)
Comment (zentol) [reply flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java:106]: IMO we shouldn't deprecate something when at the time of deprecation there is no alternative. Effectively you're forcing 1.15 users to use deprecated APIs.

Comment (dannycranmer) [reply flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java:106]: I agree

Comment (CrynetLogistics) [reply flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumer.java:106]: I couldn't agree more

**Review by dannycranmer — APPROVED** [review]

**Comment by dannycranmer** [comment]
LGTM, merging

**Pull request merged**